### PR TITLE
Issue #446: Fixed calls to open without ctx manager

### LINKED ISSUE #446: Fix calls to `open` without context manager
Run

```
pipx run ruff check --select SIM115
```

to find the issues.

Ideally replace with 

```
Path(...).read_text()
```

or if that doesn't work as described here: https://docs.astral.sh/ruff/rules/open-file-with-context-handler/

### PR DESCRIPTION
**Pull request by milaiwi**
#### Reference Issues/PRs

Fixes #446 

#### What does this implement/fix? Explain your changes.

Fixes issues identified by Ruff which advises on proper file handling in Python. Replaces direct file handling methods with safer context handler by using `Path.read_text()` and `with open(...) as file:` where appropriate. 

#### Files

1. evaluation/aggregate_results.py
2. evaluation/evaluation.py
3. inspector/server.py
4. make_demos/convert_traj_to_demo.py
5. run_replay.py
6. sweagent/agent/agents.py
7. sweagent/agent/commands.py
8. sweagent/agent/models.py
9. sweagent/environment/swe_env.py
10. sweagent/environment/utils.py
11. inspector/static.py

### CODE REVIEW

**Comment by klieret** [comment]
Just skimmed it and it looks good. Don't mind the build-docs failure, that's probably on me.

Gonna look a bit more thoroughly in a bit!

File: inspector/server.py [hunk 1]
```diff
@@ -277,15 +277,18 @@ def main(data_path, directory, port):
     data = []
     if data_path is not None:
         if data_path.endswith(".jsonl"):
-            data = [json.loads(x) for x in open(data_path).readlines()]
+            data = [json.loads(x) for x in Path(data_path).read_text().splitlines(keepend=True)]
```
Comment (klieret) [inline inspector/server.py:280]: this should be `keepends`

**Review by klieret — CHANGES_REQUESTED** [review]

File: inspector/server.py (hunk 1 above)
Comment (klieret) [reply inspector/server.py:280]: Will merge after this is resolved :)

Comment (milaiwi) [reply inspector/server.py]: Fixed! :)

**Comment by ofirpress** [comment]
Thanks @milaiwi !

**Comment by klieret** [comment]
Thanks!

**Pull request merged**